### ISSUE TITLE
Question: from custom segmentation to custom diarization model

### ISSUE THREAD

**Issue by IzzyHibbert** [pyannote/pyannote-audio#1748]
### Tested versions

Hi.
Now that I completed a **custom segmentation model** (based on segmentation 3.0, trained on my dataset), I read that the segmentation model only processes 10 seconds chunks, so to perform full recording one should make the **diarization model** which uses another speaker embedding.
I went through the details and tutorials but I am not sure I founded the part where with a new, custom segmentation model, you make the new diarization model. 
Thanks

### System information

not applicable.

### Issue description

Just need to clarify how to use a custom segmentation model to get a new diarization model that uses that.

### Minimal reproduction example (MRE)

(question)

**Comment by FrenchKrab** [comment]
If I understand correctly, you just need to plug your local segmentation model into a pipeline.
See https://github.com/pyannote/pyannote-audio/blob/develop/tutorials/adapting_pretrained_pipeline.ipynb.

What you've done is probably section "Fine-tuning the segmentation model", now you have to do "Optimizing the pipeline hyper-parameters" (you do not *have* to optimize the hyperparameters but it will show you how to create/initialize the pipeline object).
Basically check out the parameters of the SpeakerDiarization pipeline class: https://github.com/pyannote/pyannote-audio/blob/286ea1a4e34e2dd7d7926f590e402dac1e17494b/pyannote/audio/pipelines/speaker_diarization.py#L115-L126

You can check the pipeline on huggingface to see what are the latest hparams used by pyannote: https://huggingface.co/pyannote/speaker-diarization-3.1/blob/main/config.yaml

**Comment by IzzyHibbert** [comment]
> t _have_ to optimize the hyperparameters but it will show you how to create/initialize the pipeline object)

Hi.
Thanks a lot for the prompt and clear response.
In my case, the training was made through the diarizers: 
[https://github.com/huggingface/diarizers/tree/main?tab=readme-ov-file#train](https://github.com/huggingface/diarizers/tree/main?tab=readme-ov-file#train)

I guess now I am missing the hyper-parameters optimization step you suggested and can take it from there trough the adapting pretrained pipeline.. Am I correct ?

**Comment by IzzyHibbert** [comment]
> If I understand correctly, you just need to plug your local segmentation model into a pipeline. See https://github.com/pyannote/pyannote-audio/blob/develop/tutorials/adapting_pretrained_pipeline.ipynb.
> 
> What you've done is probably section "Fine-tuning the segmentation model", now you have to do "Optimizing the pipeline hyper-parameters" (you do not _have_ to optimize the hyperparameters but it will show you how to create/initialize the pipeline object). Basically check out the parameters of the SpeakerDiarization pipeline class:
> 
> https://github.com/pyannote/pyannote-audio/blob/286ea1a4e34e2dd7d7926f590e402dac1e17494b/pyannote/audio/pipelines/speaker_diarization.py#L115-L126
> 
> You can check the pipeline on huggingface to see what are the latest hparams used by pyannote: https://huggingface.co/pyannote/speaker-diarization-3.1/blob/main/config.yaml

Thanks for answering.
In my scenario I'd have to use a (diarization) model, not the pipeline.
For instance I saw that the segmentation default in the **Loading models from 🤗 hub** :

[https://github.com/pyannote/pyannote-audio/blob/develop/tutorials/applying_a_model.ipynb](url)

But, if I am not wrong, there is no tutorial to create a final diarization model, once you have your segmentation one ready.

I am trying to use the [whisper-plus](https://github.com/kadirnar/whisper-plus) project, ideally wanted to simply swap diarizer_model that easy.
Anything helps to create the final diarizer model ?

**Comment by FrenchKrab** [comment]
I don't think there is a "final" diarization model. If your goal is whisper-plus, in the examples of the README they're using `diarizer_model="pyannote/speaker-diarization-3.1",` which is a pipeline (the local eend segmentation model is applied with a sliding window to cover the file and then aggregated through embedding extraction + clustering, as described in the pyannote.audio 2.1 paper).

So I think you can follow my previous instructions to initialize the pipeline and plug in your own segmentation model, and then plug that pipeline into the whisper-plus with something like `your_whisper_pipeline.diarization_pipeline = your_pyannoteaudio_pipeline` (https://github.com/kadirnar/whisper-plus/blob/568308a2971c7c1ef424f5cec9d28a02b13e9a24/whisperplus/pipelines/whisper_diarize.py#L43)

**Comment by IzzyHibbert** [comment]
> I don't think there is a "final" diarization model. If your goal is whisper-plus, in the examples of the README they're using `diarizer_model="pyannote/speaker-diarization-3.1",` which is a pipeline (the local eend segmentation model is applied with a sliding window to cover the file and then aggregated through embedding extraction + clustering, as described in the pyannote.audio 2.1 paper).
> 
> So I think you can follow my previous instructions to initialize the pipeline and plug in your own segmentation model, and then plug that pipeline into the whisper-plus with something like `your_whisper_pipeline.diarization_pipeline = your_pyannoteaudio_pipeline` (https://github.com/kadirnar/whisper-plus/blob/568308a2971c7c1ef424f5cec9d28a02b13e9a24/whisperplus/pipelines/whisper_diarize.py#L43)

Thanks so much for taking the time and pointing me to the solution. Appreciated. 👍🏻
